packages/commonwealth/server/routes/getSnapshotProposal.ts

```diff
@@ -0,0 +1,40 @@
+import { Response } from 'express';
+import { DB } from '../models';
+import { success, TypedRequestQuery } from '../types';
+import fetchNewSnapshotProposal from '../util/fetchSnapshot'
+
+export const Errors = {
+  NoId: 'No id was provided, cannot fetch snapshot proposal',
+};
+
+type GetSnapshotProposalReq = { id: string };
+
+const getSnapshotProposal = async (
+  models: DB,
+  req: TypedRequestQuery<GetSnapshotProposalReq>,
+  res: Response,
+) => {
+  try {
+    const { id } = req.query;
+    if (!id) {
+      return res.status(400).json({ error: Errors.NoId });
+    }
+
+    const proposal = await models.SnapshotProposal.findOne({
+      where: { id },
+    });
+
+
+    if (!proposal) {
+      const createdProposal = await fetchNewSnapshotProposal(id, models);
```

Comment:
So -- is this a hack around if the listener isn't working properly?

Reply:
This is for testing locally.